csharp/src/Ice/OpaqueEndpoint.cs

```diff
@@ -111,6 +105,13 @@ protected internal override void AppendOptions(StringBuilder sb, char optionSepa
             }
         }
 
+        protected internal override Endpoint Clone(string host) =>
```

Comment:
Can we just throw `InvalidOperationException`, seems this is only used to compute the OA published endpoints and it cannot contain opaque endpoints.

Reply:
I think we should keep the implementation. In the future we may find another use case.

Reply:
Added a TODO.

Reply:
Removed TODO and replaced Clone by GetPublishedEndpoint that throws NotSupportedException as suggested by Jose/Joe.